tests/conftest.py

```diff
@@ -65,35 +65,52 @@ def small_sky_parts_dir_local(local_data_dir):
 
 
 @pytest.fixture
-def tmp_dir_cloud(example_cloud_path):
-    return os.path.join(example_cloud_path, "tmp")
+def test_data_dir_cloud(cloud_path):
+    return os.path.join(cloud_path, "data")
 
 
 @pytest.fixture
-def test_data_dir_cloud(example_cloud_path):
-    return os.path.join(example_cloud_path, "data")
+def almanac_dir_cloud(cloud_path):
+    return os.path.join(cloud_path, "data", ALMANAC_DIR_NAME)
 
 
 @pytest.fixture
-def almanac_dir_cloud(example_cloud_path):
-    return os.path.join(example_cloud_path, "data", ALMANAC_DIR_NAME)
+def small_sky_dir_cloud(cloud_path):
+    return os.path.join(cloud_path, "data", SMALL_SKY_DIR_NAME)
 
 
 @pytest.fixture
-def small_sky_dir_cloud(example_cloud_path):
-    return os.path.join(example_cloud_path, "data", SMALL_SKY_DIR_NAME)
+def small_sky_order1_dir_cloud(cloud_path):
+    return os.path.join(cloud_path, "data", SMALL_SKY_ORDER1_DIR_NAME)
 
 
 @pytest.fixture
-def small_sky_order1_dir_cloud(example_cloud_path):
-    return os.path.join(example_cloud_path, "data", SMALL_SKY_ORDER1_DIR_NAME)
+def small_sky_index_dir_cloud(cloud_path):
+    return os.path.join(cloud_path, "data", "small_sky_object_index")
 
 
 @pytest.fixture
-def small_sky_index_dir_cloud(example_cloud_path):
-    return os.path.join(example_cloud_path, "data", "small_sky_object_index")
+def small_sky_margin_dir_cloud(cloud_path):
+    return os.path.join(cloud_path, "data", "small_sky_order1_margin")
+
+
+@pytest.fixture(scope="session", name="tmp_dir_cloud")
+def tmp_dir_cloud(cloud_path, storage_options):
+    """Create a single client for use by all unit test cases."""
+    # client = Client()
+    tmp = TempCloudDirectory(
+        os.path.join(cloud_path, "tmp"),
+        method_name="full_test",
+        storage_options=storage_options,
+    )
+    yield tmp.open()
```

Comment:
Will this do a new open each time `tmp_dir_cloud` is called?

I was expecting something like:
```
path = tmp.open()
yield path
```

but I'm not actually sure how this works in Python.
```

Reply:
Short answer: no.

Longer answer: some kinda python magic makes sure this is only called once, and by using `yield`, we can clean up whatever state we need to after the `yield` call. There's some caching happening, and I don't fully understand.

But in testing, I did confirm that it's all using the same opened directory.